packages/fields/src/fields/author/index.tsx

```diff
@@ -35,6 +35,7 @@ const authorField: Field< BasePostWithEmbeddedAuthor > = {
 			label: name,
 		} ) );
 	},
+	setValue: ( { value } ) => ( { author: Number( value ) } ),
```

Comment:
Since `author` value is an integer we should be a bit more defensive. The reason I added here is because when we edit through the `select` control, the value becomes a `string`. There is no problem with REST API, but I'd have to coerce the values to the same type in `AuthorView` when I'm comparing `item.author` and `item._embedded.author` ids.